Tests/RevenueCatUITests/CustomerCenter/ManageSubscriptionsViewModelTests.swift

```diff
@@ -196,6 +196,139 @@ class ManageSubscriptionsViewModelTests: TestCase {
         expect(subscriptionInformation.productIdentifier) == productIdOne
     }
 
+    func testShouldShowEarliestExpiration_whenUserHasLifetimeAndSubscriptionsOneEntitlement() async throws {
+        let productIdLifetime = "com.revenuecat.simpleapp.lifetime"
+        let productIdMonthly = "com.revenuecat.simpleapp.monthly"
+        let productIdYearly = "com.revenuecat.simpleapp.yearly"
+        let purchaseDateLifetime = "2024-11-21T16:04:20Z"
+        let purchaseDateMonthly = "2024-11-21T16:04:39Z"
+        let purchaseDateYearly = "2024-11-21T16:04:45Z"
+        let expirationDateMonthly = "2024-11-28T16:04:39Z"
+        let expirationDateYearly = "2025-11-21T16:04:45Z"
+
+        let products = [
+            SubscriptionInformationFixtures.product(id: productIdLifetime,
+                                                    title: "lifetime",
+                                                    duration: nil,
+                                                    price: 29.99),
+            SubscriptionInformationFixtures.product(id: productIdMonthly,
+                                                    title: "monthly",
+                                                    duration: .month,
+                                                    price: 2.99),
+            SubscriptionInformationFixtures.product(id: productIdYearly,
+                                                    title: "yearly",
+                                                    duration: .year,
+                                                    price: 29.99)
+        ]
+
+        let customerInfo = CustomerInfoFixtures.customerInfo(
+            subscriptions: [
+                CustomerInfoFixtures.Subscription(
+                    id: productIdMonthly,
+                    store: "app_store",
+                    purchaseDate: purchaseDateMonthly,
+                    expirationDate: expirationDateMonthly
+                ),
+                CustomerInfoFixtures.Subscription(
+                    id: productIdYearly,
+                    store: "app_store",
+                    purchaseDate: purchaseDateYearly,
+                    expirationDate: expirationDateYearly
+                )
+            ].shuffled(),
```

Comment:
Instead of using `.shuffled()`, you might consider making this a parameterized test and passing in `[monthlyProduct, yearlyProduct]` and `[yearlyProduct, monthlyProduct]` separately.

This way, we always test both cases instead of randomly testing one case

Reply:
good idea. I did my own version of a parameterized test by making sure both cases get always tested. I don't think we have any other parameterized test in the source code and was spending too much time trying to build somehting generic lol

Reply:
We have this existing parameterized test 😄 

[`CustomerCenterViewModelTests.testAppIsLatestVersion()`](https://github.com/RevenueCat/purchases-ios/blob/4a0156e1f63fb41c975d8df9d1e819a3ab3e7deb/Tests/RevenueCatUITests/CustomerCenter/CustomerCenterViewModelTests.swift#L129)